autoprecompiles/src/powdr.rs

```diff
@@ -219,37 +164,37 @@ impl<'a, T: Clone + Ord + std::fmt::Display + 'a, E: AllChildren<AlgebraicExpres
     }
 }
 
-pub fn reassign_ids_algebraic<T: Clone + Ord>(
+/// Given an expression `expr`
+/// - Reassign the IDs of all references in the expression to be unique and larger or equal than `curr_id`
+/// - Add a suffix to the name of each reference, so that `foo` becomes `foo_suffix`
```

Comment:
I kinda feel like these should be kept separate, but fine to try this first and separate as needed in the future.